locking/locks.go

```diff
@@ -22,31 +21,51 @@ var (
 	ErrLockAmbiguous = errors.New("lfs: multiple locks found; ambiguous")
 )
 
+type LockCacher interface {
+	Add(l Lock) error
+	RemoveByPath(filePath string) error
+	RemoveById(id string) error
+	Locks() []Lock
+	Clear()
+	Save() error
+}
+
 // Client is the main interface object for the locking package
 type Client struct {
-	cfg       *config.Configuration
-	apiClient *api.Client
-	cache     *LockCache
+	Remote string
+	client *lockClient
+	cache  LockCacher
 }
 
 // NewClient creates a new locking client with the given configuration
 // You must call the returned object's `Close` method when you are finished with
 // it
-func NewClient(cfg *config.Configuration) (*Client, error) {
-
-	apiClient := api.NewClient(api.NewHttpLifecycle(cfg))
+func NewClient(remote string, lfsClient *lfsapi.Client) (*Client, error) {
+	return &Client{
+		Remote: remote,
+		client: &lockClient{Client: lfsClient},
+		cache:  &nilLockCacher{},
```

Comment:
I'm not sure I'm a fan of this pattern in Go. I'd rather sacrifice some if-statements to have a more straightforward set of types than have gain an interface and another implementation. I don't feel strongly, so if you like this pattern, 👍.